pallets/automation-time/src/lib.rs

```diff
@@ -1014,6 +1043,44 @@ pub mod pallet {
 			(task, <T as Config>::WeightInfo::run_auto_compound_delegated_stake_task())
 		}
 
+		/// Attempt to decode and run the call.
+		pub fn run_dynamic_dispatch_action(
+			caller: AccountOf<T>,
+			encoded_call: Vec<u8>,
+			task_id: TaskId<T>,
+		) -> Weight {
+			match <T as Config>::Call::decode(&mut &*encoded_call) {
+				Ok(scheduled_call) => {
+					let dispatch_origin: T::Origin =
+						frame_system::RawOrigin::Signed(caller.clone()).into();
+
+					let call_weight = scheduled_call.get_dispatch_info().weight;
+					let (maybe_actual_call_weight, result) =
+						match scheduled_call.dispatch(dispatch_origin) {
+							Ok(post_info) => (post_info.actual_weight, Ok(())),
+							Err(error_and_info) =>
+								(error_and_info.post_info.actual_weight, Err(error_and_info.error)),
+						};
+
+					Self::deposit_event(Event::DynamicDispatchResult {
```

Comment:
You can see an example in the tests. 
```
				[Event::AutomationTime(crate::Event::DynamicDispatchResult {
					who: account_id,
					task_id,
					result: Ok(()),
				}),]
```

is that sufficient?

Reply:
I'm more interested in how the error gets serialized. Hard to see that from the test. Also, I'm curious what the event would look like if you used the original result of the dispatch rather than just `Ok(())`. I suspect that we're throwing away information that might be relevant for some dispatches.

Reply:
The answer I'm trying to answer is what is the advantage of mapping to a new result with less info rather than just
```
Event::DynamicDispatchResult {
  who: account_id,
  task_id,
  result: scheduled_call.dispatch(dispatch_origin), 
}
```

Reply:
Ah I see the confusion. `PostDispatchInfo` only contains [fee related data](https://github.com/paritytech/substrate/blob/master/frame/support/src/weights.rs#L307). There is no way to return application specific information. That is why we just return `OK(())`. 

An event with an error:

Reply:
<img width="493" alt="Screen Shot 2022-08-29 at 3 27 45 PM" src="https://user-images.githubusercontent.com/9358087/187309987-076417fc-65a3-4fad-88de-96c7fdfa7dce.png">

Reply:
Ah, thanks for clarifying. Digging into this it looks like it's built to be able to return anything that implements the PostInfo type but for now it seems that's only being used for weight like you said.
https://github.com/paritytech/substrate/blob/5cdb054049980eecb449c36a7ccfe4990329c494/primitives/runtime/src/traits.rs#L1220-L1224